### ISSUE TITLE
iOS Build Failing, work arounds in previous bugs not working.

### ISSUE THREAD

**Issue by lfg-ryan** [mopub/mopub-unity-sdk#171]
iOS Build Failing "Assets/MoPub/Plugins/iOS/MoPubManager.h:15:9: 'MoPub.h' file not found".  Tried several work around listed in closed tickets including !use_frameworks removal and

- [x] I am submitting a bug report for existing functionality
- [x] I visited https://developers.mopub.com/ and found no answer
- [x] I checked https://twittercommunity.com/c/advertiser-api/mopub to make sure that this issue has not already been filed
- [x] I checked to make sure that this issue has not already been filed

#### Steps to reproduce the behavior:
Integrated MoPub Unity SDK, Added Mediation for almost all networks, iOS builds do not work locally or through Unity Cloud build.

#### Expected behavior:
Builds should succeed.

#### Observed behavior:
Out of the box, the build fails with:
`Assets/MoPub/Plugins/iOS/MoPubManager.h:15:9: 'MoPub.h' file not found`

If I do the suggested removal of !use_frameworks work around it fails with:
```
141314: ⚠;️  ld: Could not find or use auto-linked library 'swiftObjectiveC'
141315: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreFoundation'
141316: ⚠;️  ld: Could not find or use auto-linked library 'swiftCompatibility50'
141317: ⚠;️  ld: Could not find or use auto-linked library 'swiftDispatch'
141318: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreGraphics'
141319: ⚠;️  ld: Could not find or use auto-linked library 'swiftUIKit'
141320: ⚠;️  ld: Could not find or use auto-linked library 'swiftMetal'
141321: ⚠;️  ld: Could not find or use auto-linked library 'swiftWebKit'
141322: ⚠;️  ld: Could not find or use auto-linked library 'swiftAVFoundation'
141323: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreMedia'
141324: ⚠;️  ld: Could not find or use auto-linked library 'swiftCompatibilityDynamicReplacements'
141325: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreImage'
141326: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreAudio'
141327: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreMIDI'
141328: ⚠;️  ld: Could not find or use auto-linked library 'swiftUniformTypeIdentifiers'
141329: ⚠;️  ld: Could not find or use auto-linked library 'swiftDarwin'
141330: ⚠;️  ld: Could not find or use auto-linked library 'swiftMediaPlayer'
141331: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreLocation'
141332: ⚠;️  ld: Could not find or use auto-linked library 'swiftQuartzCore'
141333: ⚠;️  ld: Could not find or use auto-linked library 'swiftCore'
141334: ⚠;️  ld: Could not find or use auto-linked library 'swiftCompatibility51'
141335: ⚠;️  ld: Could not find or use auto-linked library 'swiftsimd'
141336: ⚠;️  ld: Could not find or use auto-linked library 'swiftFoundation'
141337: ❌;  ld: symbol(s) not found for architecture arm64
141338: ❌;  clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

If I do the multi step process of removing a dependency line from MoPubDependencies.xml, deleting the podspec file and including all files in the Plugins/iOS folder work around It fails with:

```
70946: ⚠;️  ld: Could not find or use auto-linked library 'swiftObjectiveC'
70947: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreFoundation'
70948: ⚠;️  ld: Could not find or use auto-linked library 'swiftCompatibility50'
70949: ⚠;️  ld: Could not find or use auto-linked library 'swiftDispatch'
70950: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreGraphics'
70951: ⚠;️  ld: Could not find or use auto-linked library 'swiftUIKit'
70952: ⚠;️  ld: Could not find or use auto-linked library 'swiftMetal'
70953: ⚠;️  ld: Could not find or use auto-linked library 'swiftWebKit'
70954: ⚠;️  ld: Could not find or use auto-linked library 'swiftAVFoundation'
70955: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreMedia'
70956: ⚠;️  ld: Could not find or use auto-linked library 'swiftCompatibilityDynamicReplacements'
70957: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreImage'
70958: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreAudio'
70959: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreMIDI'
70960: ⚠;️  ld: Could not find or use auto-linked library 'swiftUniformTypeIdentifiers'
70961: ⚠;️  ld: Could not find or use auto-linked library 'swiftDarwin'
70962: ⚠;️  ld: Could not find or use auto-linked library 'swiftMediaPlayer'
70963: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreLocation'
70964: ⚠;️  ld: Could not find or use auto-linked library 'swiftQuartzCore'
70965: ⚠;️  ld: Could not find or use auto-linked library 'swiftCore'
70966: ⚠;️  ld: Could not find or use auto-linked library 'swiftCompatibility51'
70967: ⚠;️  ld: Could not find or use auto-linked library 'swiftsimd'
70968: ⚠;️  ld: Could not find or use auto-linked library 'swiftFoundation'
70969: ❌;  ld: symbol(s) not found for architecture arm64
70970: ❌;  clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

#### Evidence:
N/A

**Comment by hyeonc** [comment]
I had exactly the same problem.
I checked that 'unity-sample-app' is running normally.
So I overwritten my project with the 'Assets/MoPub' directory included in 'unity-sample-app'.
After that, it worked.
(Mopub Version is 5.15.0 and Unity Version is 2019.4.21f1)

**Comment by lfg-ryan** [comment]
Thanks for the suggestion. Tried dragging that folder over. Git shows no delta except the inclusion of an empty MoPubSDKVersion.cs so I don't think this is the issue.

**Comment by lfg-ryan** [comment]
to be clear all mediation layers are installed except Pangle and Verizon. They are on the latest versions as of the date of the original bug post. (looks like ironsource has a new 1.2.7 that is not yet upgraded).

**Comment by dravyan** [comment]
Thank you for the report. We will take a look and get back to you. There is workaround suggested in the [earlier report](https://github.com/mopub/mopub-unity-sdk/issues/151). Let us know if that helps.
Also, could you confirm if you have any other third-party Unity SDKs integrated into your project alongside MoPub Unity SDK and adapters?

Thanks,
MoPub Team

**Comment by lfg-ryan** [comment]
Thank you. As mentioned in the bug I tried that work around and that's what lead to a new failure 

```
141314: ⚠;️  ld: Could not find or use auto-linked library 'swiftObjectiveC'
141315: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreFoundation'
141316: ⚠;️  ld: Could not find or use auto-linked library 'swiftCompatibility50'
141317: ⚠;️  ld: Could not find or use auto-linked library 'swiftDispatch'
141318: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreGraphics'
141319: ⚠;️  ld: Could not find or use auto-linked library 'swiftUIKit'
141320: ⚠;️  ld: Could not find or use auto-linked library 'swiftMetal'
141321: ⚠;️  ld: Could not find or use auto-linked library 'swiftWebKit'
141322: ⚠;️  ld: Could not find or use auto-linked library 'swiftAVFoundation'
141323: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreMedia'
141324: ⚠;️  ld: Could not find or use auto-linked library 'swiftCompatibilityDynamicReplacements'
141325: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreImage'
141326: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreAudio'
141327: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreMIDI'
141328: ⚠;️  ld: Could not find or use auto-linked library 'swiftUniformTypeIdentifiers'
141329: ⚠;️  ld: Could not find or use auto-linked library 'swiftDarwin'
141330: ⚠;️  ld: Could not find or use auto-linked library 'swiftMediaPlayer'
141331: ⚠;️  ld: Could not find or use auto-linked library 'swiftCoreLocation'
141332: ⚠;️  ld: Could not find or use auto-linked library 'swiftQuartzCore'
141333: ⚠;️  ld: Could not find or use auto-linked library 'swiftCore'
141334: ⚠;️  ld: Could not find or use auto-linked library 'swiftCompatibility51'
141335: ⚠;️  ld: Could not find or use auto-linked library 'swiftsimd'
141336: ⚠;️  ld: Could not find or use auto-linked library 'swiftFoundation'
141337: ❌;  ld: symbol(s) not found for architecture arm64
141338: ❌;  clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by lfg-ryan** [comment]
After trying various combinations of previous bugs, the closest I seem to get is still linking issues.

This is the latest

"_UnityPause", referenced from: -[MoPubManager willPresentModalViewForAd:] in libMoPub-SDK-Plugin.a(MoPubManager.o)

At this point I have made so many modifications to project settings and plists, its getting hard to sort through. Here are the modifications I am doing in post build steps.

1 - Modifying `GCC_ENABLE_OBJC_EXCEPTIONS, GCC_ENABLE_CPP_EXCEPTIONS, CLANG_ENABLE_MODULE`S from No to Yes
2 - Updating swift library search paths to `$(SDKROOT)/usr/lib/swift $(TOOLCHAIN_DIR)/usr/lib/swift/$(PLATFORM_NAME) $(TOOLCHAIN_DIR)/usr/lib/swift-5.0/$(PLATFORM_NAME)`
3 - Excluding archs - `i386, x86_64`
4 - Removing  `use_frameworks!` from the podfile and rerunning `pod update`

I feel like I have exhausted the different combinations of previously mentioned work arounds.

**Comment by dravyan** [comment]
@lfg-ryan, Could you let us know if you are integrating any other third-party SDKs/Framework outside of MoPub integration? . Is it possible to share your project zip to investigate this further?

**Comment by lfg-ryan** [comment]
Ok I got it working. Here were my modifications for iOS.

Note: I was using the ExternalDependencyManager through the Package manager which was using version 1.2.164, Mopub was previously using 1.2.160. In 164 EDM added some new podfile config options that seem to be at least part of the problem.

My setup.

1) Manually include the iOS framework into my Plugins/iOS directory. .framework file grabbed from this. https://github.com/mopub/mopub-ios-sdk/releases/download/5.16.1/mopub-framework-5.16.1.zip, unpack the zip, go into the MoPubSDK.xcframework and grabe the MoPubSDK.framework under ios-arm64_armv7 (note i don't care about simulator, so didn't try using the xcframework directly).

2) Do the suggested change that other bugs have mentioned.

- Remove <iosPod name="MoPub-SDK-Plugin" minTargetSdk="10.0" path="Assets/MoPub/Plugins/iOS"/> from MoPubDependencies.xml
- Delete Assets/MoPub/Plugins/iOS/MoPub-SDK-Plugin.podspec
- Check iOS as platform for plugin for all files inside Assets/MoPub/Plugins/iOS

3) (if using EDM 1.2.164 or greater) go to Assets > External Depedency Manager > iOS Resolver > Settings and uncheck the 2 options under Podfile Configurations. I think the "Add use_frameworks! to Podfile" is most critical. No idea why EDM has that as the default.
<img width="1249" alt="Screen Shot 2021-03-12 at 5 06 34 PM" src="https://user-images.githubusercontent.com/49963806/111013712-4f403980-8355-11eb-9b7a-a392be3be7e5.png">

**Comment by lfg-ryan** [comment]
ok, not out of the woods yet. While it builds it crashes on launch.

dependent dylib '@rpath/MoPubSDK.framework/MoPubSDK' not found

**Comment by dravyan** [comment]
@lfg-ryan , since you integrated MoPub SDK manually, you may need to include the `MoPubSDK.framework` under `Embed Frameworks` in Xcode build settings.

**Comment by lfg-ryan** [comment]
In Unity. Selecting the MoPubSDK.framework and checking the box for "Add to Embedded Binaries" did the trick. App now launches.
![Screen Shot 2021-03-16 at 12 51 55 PM](https://user-images.githubusercontent.com/49963806/111371363-887af100-8656-11eb-8700-825580d2a4a5.png)

**Comment by dravyan** [comment]
This issue has been resolved in [5.16.2](https://github.com/mopub/mopub-unity-sdk/releases/tag/v5.16.2) SDK release. Please update.

Thanks,
MoPub Team
